socorro/external/es/supersearch.py

```diff
@@ -99,19 +99,20 @@ def format_field_names(self, hit):
     def format_fields(self, hit):
         """Return a well formatted document.
 
-        Elasticsearch returns values as lists when using the `fields` option.
-        This function removes the list when it contains zero or one element.
-        It also calls `format_field_names` to correct all the field names.
+        Elasticsearch returns a nested document, just like it is stored in
+        the database. We want to flatten that out.
         """
-        hit = self.format_field_names(hit)
-
-        for field in hit:
-            if isinstance(hit[field], (list, tuple)):
-                if len(hit[field]) == 0:
-                    hit[field] = None
-                elif len(hit[field]) == 1:
-                    hit[field] = hit[field][0]
+        def flatten(d, parent_key='', sep='.'):
+            items = []
+            for k, v in d.items():
+                new_key = parent_key + sep + k if parent_key else k
+                if isinstance(v, collections.MutableMapping):
+                    items.extend(flatten(v, new_key, sep=sep).items())
+                else:
+                    items.append((new_key, v))
+            return dict(items)
```

Comment:
You're right, so I started moving it to ``socorro/lib/util.py``, and I feel like I've open Pandora's box... That file is a huge mess, I'm trying to clean it up, that will add a bunch of code changes to this PR but it will be in a separate commit.